pandas/core/config_init.py

```diff
@@ -460,6 +462,40 @@ def use_inf_as_na_cb(key):
 # Plotting
 # ---------
 
+plotting_backend_doc = """
```

Comment:
One question: If I wanted to use the altar backend, I would be more like to use `.set_option('plotting.backend', 'altair')`  than `..., 'pdvega')`. @jakevdp what name would you prefer?

I think hvplot will just be hvplot, so that's fine.

Anyway, we might consider adding a dict here like `plotting_backend_alias` that maps the user-facing name like `altair` to the backend name like `pdvega`. When the backend library registers themselves, they can also register their aliases.

Reply:
I see your point, and I think it'd add value to the users, but not sure if I'm in favor of adding the extra complexity it's needed to manage aliases in a dynamic way.

I like the simplicity of the parameter being the name of the module. I guess in some cases will look nicer than others. May be `hvplot` will use `hvplot.pandas`, since hvplot contains other things besides our plugin, and the module to use may be `hvplot.pandas`.

In practice I guess backends will register themselves, and users will rarely switch backends manually. But I guess if they do, it'll be better if they know they need to use the name of the module:
```python
import pandas
import hvplot.pandas

df.plot()

pandas.set_option('backend.plotting', 'matplotlib')
df.plot()

pandas.set_option('backend.plotting', 'hvplot.pandas')
df.plot()
```

I don't have a strong opinion, but I'd say let's start with the simplest option, and add aliases or something else if we think it's useful once we start using this.

Reply:
Is it especially complex? I was thinking something like

```python
_plotting_aliases = {}  # or define somewhere else

def register_plotting_backend_cb(key):
    backend_str = cf.get_option(key)
    backend_str = _plotting_aliases.get(backend_str, backend_str)
```

Indeed, I think this simplifies things already, since we can use `'matplotlib'`  as `pandas.plotting._matplotlib`. Though we may continue to special case matplotlib to provide a nice error message.

Reply:
What I wouldn't do is to have the aliases in pandas itself. May be I'm being too strict, but if feels wrong.

But you're right, it's probably not as complex as I was thinking anyway. An simple option `plotting.aliases` with a dictionary may not be ideal, but would allow backends create an alias by simply:
```python
pandas.set_option('plotting.aliases', dict(pandas.get_option('plotting.aliases'), hbplot='hvplot.pandas'))
```

Reply:
but better in a follow up PR I think, so we can focus there on the exact syntax and approach

Reply:
Perfectly fine doing as a followup.

And my thinking may have been a bit muddled here. I was thinking that the backend library would have already been imported, and so would have a chance to register their own aliases. But as you say, it would be pandas managing them, which doesn't feel quite right.

Reply:
another simple option is that backends add an optional attribute `alias = 'hvplot'`, and we simply do:
```python
if hasattr(backend_mod, 'alias'):
    plotting_aliases[alias] = backend_mod.__name__
```

Reply:
Yes good idea. But still leaving this as a followup?

Reply:
Yes, I prefer to keep the focus, the smaller the PRs, the better the content :)